### PR TITLE
Include Multi namespace Invoke support

### LINKED ISSUE #69: Add multi-namespace support to the invoke test cases
Refactor the test cases in the `invoke_test.go` to use the same table driven test pattern introduced in #66 so that we can run the tests in the default "empty" namespace and a non-default namespace when it is configured. 

Part of #61

### PR DESCRIPTION
**Pull request by nitishkumar71**
Resolves #69

### CODE REVIEW

File: tests/function_helpers_test.go [hunk 1]
```diff
@@ -34,8 +35,8 @@ func deploy(t *testing.T, createRequest *sdk.DeployFunctionSpec) int {
 	return statusCode
 }
 
-func list(t *testing.T, expectedStatusCode int) {
-	functions, err := config.Client.ListFunctions(context.Background(), defaultNamespace)
+func list(t *testing.T, expectedStatusCode int, optional ...interface{}) {
```
Comment (alexellis) [inline tests/function_helpers_test.go:38]: Can we avoid using this untyped parameter `... interface{}`? What is it doing?

Comment (nitishkumar71) [reply tests/function_helpers_test.go]: Thanks for pointing it out.

File: tests/function_helpers_test.go [hunk 2]
```diff
@@ -76,3 +77,19 @@ func scaleFunction(t *testing.T, name string, count int) {
 		t.Fatalf("scale got %d, wanted %d (or %d)", res.StatusCode, http.StatusAccepted, http.StatusOK)
 	}
 }
+
+func copyNamespacesTest(cases []FunctionMetaSchema) []FunctionMetaSchema {
```
Comment (LucasRoesler) [inline tests/function_helpers_test.go:81]: this is a nitpick, but i wonder if something like `FunctionTestCase` might be a better name for this type

**Review by LucasRoesler — COMMENTED** [review]
Is it possible for us to unify the various invoke test cases into a single list of test cases instead of several different test functions? I think it will make it easier to maintain because we can easily read the one list of cases for the invoke behavior that we expect

File: tests/function_helpers_test.go (hunk 2 above)
Comment (nitishkumar71) [reply tests/function_helpers_test.go:81]: The function is used to replicate testcase for another namespace. possibly correct name should be `replicateTestCaseForNamespace` with function declaration as `func replicateTestCaseforNamespace(cases []FunctionMetaSchema, namespace string) []FunctionMetaSchema`

Comment (LucasRoesler) [reply tests/function_helpers_test.go:81]: I actually mean `FunctionMetaSchema` becoming something like `FunctionTestCase`, the function name seems fine.

Comment (nitishkumar71) [reply tests/function_helpers_test.go]: @LucasRoesler refactored the code to create single list for all the test cases. Please have a look at it.

**Comment by alexellis** [comment]
@nitishkumar71 if it helps, #71 was just merged and this PR is ready for rebase.

Alex

**Comment by nitishkumar71** [comment]
> @nitishkumar71 if it helps, #71 was just merged and this PR is ready for rebase.
> 
> Alex

Sure, will rebase it asap.

**Comment by nitishkumar71** [comment]
@LucasRoesler @alexellis Please do review the changes.

File: tests/invoke_test.go [hunk 3]
```diff
@@ -66,57 +54,81 @@ func Test_Invoke_With_Supported_Verbs(t *testing.T) {
...
+
+			list(t, http.StatusOK, functionRequest.Namespace)
+
+			switch service := c.function.Service; service {
+			case "env-test-verbs":
+				invokeWithSupportedVerbs(t, functionRequest)
+			case "redirector-test":
+				_ = invoke(t, functionRequest, emptyQueryString, "", http.StatusFound)
+			case "env-test":
+				invokeWithCustomEnvVarsAndQueryString(t, functionRequest)
```
Comment (LucasRoesler) [inline tests/invoke_test.go:128]: i feel like this switch based on function names is a bit brittle and will make this harder to extend in the future. But I think we can keep it for now and address it later.

**Review by LucasRoesler — APPROVED** [review]
@nitishkumar71 sorry this took so long, i left a comment and thought i approved it _but_ i forgot to actually click the submit review button so it was still pending !!! this happens too often to me

**Comment by nitishkumar71** [comment]
No worries

**Pull request merged**